### ISSUE TITLE
Retrieving the latest `TransparencyLog` for a given artifact

### ISSUE THREAD

**Issue by dgarus** [pyrsia/pyrsia#1550]
### Suggestion details

Part of #1399 

Filter the logs to only show the latest log for a given artifact.

Query requests should be executed on the node side.

**Comment by dasmanas** [comment]
Hi @dgarus,
Can we have a little detail explanation of the issue? some suggestions here.

- Like under which scenario we have more than one log for a given artifacts.
- Which internal functions need to operate only on the latest log outcome.
- What is the criteria for selecting latest transparency log?

**Comment by dgarus** [comment]
Hello @dasmanas !

> Like under which scenario we have more than one log for a given artifacts.

When an artifact was removed or when the artifact has `num_artifacts` > 1.

> Which internal functions need to operate only on the latest log outcome.

This issue is a part of inspect-log issue, so it is more about external consumption. 
But this method can be useful to get the current state of the artifact. 
For example, if the latest log record is `RemoveArtifact` then an artifact was removed.

> What is the criteria for selecting latest transparency log?

Only `timestamp` field. There is no another way.

**Comment by dgarus** [comment]
> We need a Build ID (or similar) in the transaction log table to club these records as part of one build which will in turn help us to detect latest transparency log.

Anyway, max(timestamp) for given `package_specific_id` is the latest transparency log record. 
What is the case when this assert is wrong?

**Comment by dgarus** [comment]
@dasmanas @betarelease 

>Why do we need this feature? 

I think we could ask @erwin1, he is author of #1399.

Anyway, if we have a question like this, we should look at all issues which are parts of #1399 one more again to exclude features, that we don't need.

**Comment by betarelease** [comment]
This is limited in scope to what this story title suggests. We need a way to filter out and return only the latest log for an artifact the user is interested in.  In this context the artifact the user enters is something like `alpine-3.17.2` in which case the last log for this item should be shown. The user can use it or ask for more information using another command. But when we ask for latest using the `--latest` option the user should get the latest logs. 

For the specific case of docker where an artifact has multiple pieces to it, all of them should be returned(because the user does not care that docker has 4 files and java jar has a single file for each artifact). 

To summarize
With the pyrsia cli when the user asks for logs of a certain artifact(As specified by --image tag) with `--latest` it should return the last entries of the logs for that artifact. This behaviour is consistent with the --inspect-log behaviour as shown in the example on #1399
